src/unwinder/find_fde/custom.rs

```diff
@@ -0,0 +1,135 @@
+use super::{FDEFinder, FDESearchResult};
+use crate::util::{deref_pointer, get_unlimited_slice};
+
+use core::sync::atomic::{AtomicBool, Ordering};
+use gimli::{BaseAddresses, EhFrame, EhFrameHdr, NativeEndian, UnwindSection};
+
+static mut CUSTOM_FDE_FINDER: Option<&(dyn FDEFinder + Sync)> = None;
+
+static CUSTOM_FDE_FINDER_INIT_STARTED: AtomicBool = AtomicBool::new(false);
+static CUSTOM_FDE_FINDER_INIT_COMPLETED: AtomicBool = AtomicBool::new(false);
+
+#[derive(Debug)]
+pub struct SetCustomFDEFinderError(());
+
+pub fn set_custom_fde_finder(
+    fde_finder: &'static (dyn FDEFinder + Sync),
+) -> Result<(), SetCustomFDEFinderError> {
+    if CUSTOM_FDE_FINDER_INIT_STARTED.swap(true, Ordering::SeqCst) {
+        Err(SetCustomFDEFinderError(()))
```

Comment:
This is problematic because it can cause a case where custom FDE finder is not yet set but `Err` is returned. It should spin until the custom fde finder is set, and then return `Err`.

Reply:
Fixed.